src/script/main/app.ts

```diff
@@ -392,30 +392,30 @@ export class App {
       onProgress(10);
       telemetry.timeStep(AppInitTimingsStep.INITIALIZED_CRYPTOGRAPHY);
 
-      const team = await teamRepository.initTeam(selfUser.teamId);
+      const {members: teamMembers} = await teamRepository.initTeam(selfUser.teamId);
+      telemetry.timeStep(AppInitTimingsStep.RECEIVED_USER_DATA);
+
+      const connections = await connectionRepository.getConnections();
+      telemetry.addStatistic(AppInitStatisticsValue.CONNECTIONS, connections.length, 50);
+
+      await userRepository.loadUsers(connections, [...teamMembers, selfUser.qualifiedId]);
```

Comment:
This is the major change here. Now `getConnections` and `initTeam` do not fetch user's metadata anymore (doing less side-effects). 

The `userRepository.loadUsers` is responsible for loading the users from the DB and fetch the missing users on backend